gax-grpc/src/main/java/com/google/api/gax/grpc/ApiExceptions.java

```diff
@@ -45,20 +44,15 @@
    * following way:
    *
    * <p>1. If it is an UncheckedExecutionException, then:<br>
-   * a. If the exception cause is an ApiException, it is rethrown.<br>
-   * b. If the exception cause is a StatusRuntimeException, it is wrapped in an ApiException and
-   * thrown.<br>
+   * a. If the exception cause is an ApiException, the ApiException is rethrown.<br>
+   * b. Otherwise, the UncheckedExecutionException is rethrown.<br>
```

Comment:
This is very subtle: the interception of `StatusRuntimeException` is actually already done in the exception callable in the grpc layer, and it is converted to ApiException there. Thus, that case is subsumed in #a. I don't believe this code ever got the chance to do this unwrapping itself.